File: t/PACSFramework/VB/MedicalWorkstationCustomizationDemo/Examples/CustomizeActionsExample/ActionsEditorDialog.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports Leadtools.MedicalViewer
Imports Leadtools.Medical.Workstation.UI

Namespace Leadtools.Demos.Workstation.Customized
   Partial Public Class ActionsEditorDialog : Inherits WorkstationModalViewBase : Implements IActionsEditorView
#Region "Public"

#Region "Methods"

      Public Sub New()
         InitializeComponent()

         RegisterEvents()
      End Sub

      Public Sub SetActions(ByVal actions As MedicalViewerActionType()) Implements IActionsEditorView.SetActions
         ActionComboBox.DataSource = actions
      End Sub

      Public Sub SetActionMouseButtons(ByVal mouseButtons As MedicalViewerMouseButtons()) Implements IActionsEditorView.SetActionMouseButtons
         MouseButtonsComboBox.DataSource = mouseButtons
      End Sub

      Public Sub SetValidationErrorText(ByVal message As String) Implements IActionsEditorView.SetValidationErrorText
         errorProvider1.SetError(FeautreIdTextBox, message)
      End Sub

#End Region

#Region "Properties"

      Public Property SelectedAction() As MedicalViewerActionType Implements IActionsEditorView.SelectedAction
         Get
            Return CType(ActionComboBox.SelectedItem, MedicalViewerActionType)
         End Get

         Set(ByVal value As MedicalViewerActionType)
            ActionComboBox.SelectedItem = value
         End Set
      End Property

      Public Property ActionDisplayName() As String Implements IActionsEditorView.ActionDisplayName
         Get
            Return ActionDisplayNameTextBox.Text
         End Get

         Set(ByVal value As String)
            ActionDisplayNameTextBox.Text = value
         End Set
      End Property

      Public Property SelectedMouseButton() As MedicalViewerMouseButtons Implements IActionsEditorView.SelectedMouseButton
         Get
            Return CType(MouseButtonsComboBox.SelectedItem, MedicalViewerMouseButtons)
         End Get

         Set(ByVal value As MedicalViewerMouseButtons)
            MouseButtonsComboBox.SelectedItem = value
         End Set
      End Property

      Public Property CanEditToolstripButtons() As Boolean Implements IActionsEditorView.CanEditToolstripButtons
         Get
            Return ToolStripButtonsGroupBox.Enabled
         End Get

         Set(ByVal value As Boolean)
            ToolStripButtonsGroupBox.Enabled = value
         End Set
      End Property

      Public Property FeatureId() As String Implements IActionsEditorView.FeatureId
         Get
            Return FeautreIdTextBox.Text
         End Get

         Set(ByVal value As String)
            FeautreIdTextBox.Text = value
         End Set
      End Property

      Public Property ToolStipItemImage() As Image Implements IActionsEditorView.ToolStipItemImage
         Get
            Return ImagePictureBox.Image
         End Get

         Set(ByVal value As Image)
            ImagePictureBox.Image = value
         End Set
      End Property

      Public Property ToolStipItemAlternativeImage() As Image Implements IActionsEditorView.ToolStipItemAlternativeImage
         Get
            Return AltImagePictureBox.Image
         End Get

         Set(ByVal value As Image)
            AltImagePictureBox.Image = value
         End Set
      End Property

      Public Property CanAddAction() As Boolean Implements IActionsEditorView.CanAddAction
         Get
            Return AddActionButton.Enabled
         End Get

         Set(ByVal value As Boolean)
            AddActionButton.Enabled = value
         End Set
      End Property

      Public Property CanRemoveAction() As Boolean Implements IActionsEditorView.CanRemoveAction
         Get
            Return RemoveActionButton.Enabled
         End Get

         Set(ByVal value As Boolean)
            RemoveActionButton.Enabled = value
         End Set
      End Property

      Public Property CanChangeDisplayName() As Boolean Implements IActionsEditorView.CanChangeDisplayName
         Get
            Return ActionDisplayNameTextBox.Enabled
         End Get
         Set(ByVal value As Boolean)
            ActionDisplayNameTextBox.Enabled = value
         End Set
      End Property

      Public Property CanChangeMouseButton() As Boolean Implements IActionsEditorView.CanChangeMouseButton
         Get
            Return MouseButtonsComboBox.Enabled
         End Get
         Set(ByVal value As Boolean)
            MouseButtonsComboBox.Enabled = value
         End Set
      End Property

      Public Property CanChangeFeatureId() As Boolean Implements IActionsEditorView.CanChangeFeatureId
         Get
            Return FeautreIdTextBox.Enabled
         End Get
         Set(ByVal value As Boolean)
            FeautreIdTextBox.Enabled = value
         End Set
      End Property

#End Region

#Region "Events"

#End Region

#Region "Data Types Definition"

#End Region

#Region "Callbacks"

      Public Event SelectedActionChanged As EventHandler Implements IActionsEditorView.SelectedActionChanged
      Public Event SelectedActionMouseButtonChanged As EventHandler Implements IActionsEditorView.SelectedActionMouseButtonChanged
      Public Event AddSelectedAction As EventHandler Implements IActionsEditorView.AddSelectedAction
      Public Event RemoveSelectedAction As EventHandler Implements IActionsEditorView.RemoveSelectedAction
      Public Event ValidateFeatureId As CancelEventHandler Implements IActionsEditorView.ValidateFeatureId

#End Region

#End Region

#Region "Protected"

#Region "Methods"

#End Region

#Region "Properties"

#End Region

#Region "Events"

#End Region

#Region "Data Members"

#End Region

#Region "Data Types Definition"

#End Region

#End Region

#Region "Private"

#Region "Methods"

      Private Sub RegisterEvents()
         AddHandler ActionComboBox.SelectionChangeCommitted, AddressOf ActionComboBox_SelectionChangeCommitted
         AddHandler MouseButtonsComboBox.SelectionChangeCommitted, AddressOf MouseButtonsComboBox_SelectionChangeCommitted
         AddHandler FeautreIdTextBox.Validating, AddressOf FeautreIdTextBox_Validating
         AddHandler AddActionButton.Click, AddressOf AddActionButton_Click
         AddHandler RemoveActionButton.Click, AddressOf RemoveActionButton_Click
         AddHandler BrowseButton.Click, AddressOf BrowseButton_Click
         AddHandler AlternativeImageButton.Click, AddressOf AlternativeImageButton_Click
      End Sub

#End Region

#Region "Properties"

#End Region

#Region "Events"

      Private Sub ActionComboBox_SelectionChangeCommitted(ByVal sender As Object, ByVal e As EventArgs)
         Try
            If Not Nothing Is SelectedActionChangedEvent Then
               RaiseEvent SelectedActionChanged(Me, e)
            End If
         Catch exception As Exception
            Messager.ShowError(Me, exception)
         End Try
      End Sub

      Private Sub MouseButtonsComboBox_SelectionChangeCommitted(ByVal sender As Object, ByVal e As EventArgs)
         Try
            If Not Nothing Is SelectedActionMouseButtonChangedEvent Then
               RaiseEvent SelectedActionMouseButtonChanged(Me, e)
            End If
         Catch exception As Exception
            Messager.ShowError(Me, exception)
         End Try
      End Sub

      Private Sub FeautreIdTextBox_Validating(ByVal sender As Object, ByVal e As CancelEventArgs)
         Try
            If Not Nothing Is ValidateFeatureIdEvent Then
               RaiseEvent ValidateFeatureId(Me, e)
            End If
         Catch exception As Exception
            e.Cancel = True

            Messager.ShowError(Me, exception)
         End Try
      End Sub

      Private Sub AddActionButton_Click(ByVal sender As Object, ByVal e As EventArgs)
         Try
            If Not Nothing Is AddSelectedActionEvent Then
               RaiseEvent AddSelectedAction(Me, e)
            End If
         Catch exception As Exception
            Messager.ShowError(Me, exception)
         End Try
      End Sub

      Private Sub RemoveActionButton_Click(ByVal sender As Object, ByVal e As EventArgs)
         Try
            If Not Nothing Is RemoveSelectedActionEvent Then
               RaiseEvent RemoveSelectedAction(Me, e)
            End If
         Catch exception As Exception
            Messager.ShowError(Me, exception)
         End Try
      End Sub

      Private Sub BrowseButton_Click(ByVal sender As Object, ByVal e As EventArgs)
         Try
            Using openFileDlg As OpenFileDialog = New OpenFileDialog()
               openFileDlg.Filter = "Image Formats (*.jpg, *.bmp, *.ico)|*.jpg;*.bmp;*.ico"

               If openFileDlg.ShowDialog() = DialogResult.OK Then
                  ImagePictureBox.Image = Image.FromFile(openFileDlg.FileName)
               End If
            End Using
         Catch exception As Exception
            Messager.ShowError(Me, exception)
         End Try
      End Sub

      Private Sub AlternativeImageButton_Click(ByVal sender As Object, ByVal e As EventArgs)
         Try
            Using openFileDlg As OpenFileDialog = New OpenFileDialog()
               openFileDlg.Filter = "Image Formats (*.jpg, *.bmp, *.ico)|*.jpg;*.bmp;*.ico"

               If openFileDlg.ShowDialog() = DialogResult.OK Then
                  AltImagePictureBox.Image = Image.FromFile(openFileDlg.FileName)
               End If
            End Using
         Catch exception As Exception
            Messager.ShowError(Me, exception)
         End Try
      End Sub

#End Region

#Region "Data Members"

#End Region

#Region "Data Types Definition"

#End Region

#End Region

#Region "internal"

#Region "Methods"

#End Region

#Region "Properties"

#End Region

#Region "Events"

#End Region

#Region "Data Types Definition"

#End Region

#Region "Callbacks"

#End Region

#End Region
   End Class
End Namespace

File: t/PACSFramework/CS/MedicalWorkstationCustomizationDemo/Examples/CustomizeActionsExample/ActionsEditorDialog.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using Leadtools.MedicalViewer;
using Leadtools.Medical.Workstation.UI;

namespace Leadtools.Demos.Workstation.Customized
{
   public partial class ActionsEditorDialog : WorkstationModalViewBase, IActionsEditorView
   {
      #region Public
         
         #region Methods
         
            public ActionsEditorDialog ( )
            {
               InitializeComponent ( ) ;
               
               RegisterEvents ( ) ;
            }

            public void SetActions ( MedicalViewerActionType[] actions ) 
            {
               ActionComboBox.DataSource = actions ;
            }
            
            public void SetActionMouseButtons ( MedicalViewerMouseButtons[] mouseButtons ) 
            {
               MouseButtonsComboBox.DataSource = mouseButtons ;
            }
            
            public void SetValidationErrorText ( string message ) 
            {
               errorProvider1.SetError ( FeautreIdTextBox, message ) ;
            }
         
         #endregion
         
         #region Properties
         
            public MedicalViewerActionType SelectedAction
            {
               get 
               {
                  return (MedicalViewerActionType) ActionComboBox.SelectedItem ;
               }
               
               set
               {
                  ActionComboBox.SelectedItem = value ;
               }
            }
            
            public string ActionDisplayName
            {
               get
               {
                  return ActionDisplayNameTextBox.Text ;
               }
               
               set
               {
                  ActionDisplayNameTextBox.Text = value ;
               }
            }
            
            public MedicalViewerMouseButtons SelectedMouseButton
            {
               get
               {
                  return (MedicalViewerMouseButtons) MouseButtonsComboBox.SelectedItem ;
               }
               
               set
               {
                  MouseButtonsComboBox.SelectedItem  = value ;
               }
            }
            
            public bool CanEditToolstripButtons
            {
               get
               {
                  return ToolStripButtonsGroupBox.Enabled ;
               }
               
               set
               {
                  ToolStripButtonsGroupBox.Enabled = value ;
               }
            }
            
            public string FeatureId
            {
               get
               {
                  return FeautreIdTextBox.Text ;
               }
               
               set
               {
                  FeautreIdTextBox.Text = value ;
               }
            }
            
            public Image ToolStipItemImage
            {
               get
               {
                  return ImagePictureBox.Image ;
               }
               
               set
               {
                  ImagePictureBox.Image = value ;
               }
            }
            
            public Image ToolStipItemAlternativeImage 
            { 
               get
               {
                  return AltImagePictureBox.Image ;
               }
               
               set
               {
                  AltImagePictureBox.Image = value ;
               }
            }
            
            public bool CanAddAction
            {
               get
               {
                  return AddActionButton.Enabled ;
               }
               
               set
               {
                  AddActionButton.Enabled = value ;
               }
            }
            
            public bool CanRemoveAction
            {
               get
               {
                  return RemoveActionButton.Enabled ;
               }
               
               set
               {
                  RemoveActionButton.Enabled = value ;
               }
            }
            
            public bool CanChangeDisplayName 
            { 
               get { return ActionDisplayNameTextBox.Enabled ; }
               set { ActionDisplayNameTextBox.Enabled = value ; }
            }
            
            public bool CanChangeMouseButton 
            { 
               get { return MouseButtonsComboBox.Enabled; } 
               set { MouseButtonsComboBox.Enabled = value ; }
            }
            
            public bool CanChangeFeatureId 
            { 
               get { return FeautreIdTextBox.Enabled ; }
               set { FeautreIdTextBox.Enabled = value ; }
            }
            
         #endregion
         
         #region Events
            
         #endregion
         
         #region Data Types Definition
            
         #endregion
         
         #region Callbacks
         
            public event EventHandler SelectedActionChanged ;
            public event EventHandler SelectedActionMouseButtonChanged ;
            public event EventHandler AddSelectedAction ;
            public event EventHandler RemoveSelectedAction ;
            public event CancelEventHandler ValidateFeatureId ;
            
         #endregion
         
      #endregion
      
      #region Protected
         
         #region Methods
            
         #endregion
         
         #region Properties
            
         #endregion
         
         #region Events
            
         #endregion
         
         #region Data Members
            
         #endregion
         
         #region Data Types Definition
            
         #endregion
         
      #endregion
      
      #region Private
         
         #region Methods
         
            private void RegisterEvents ( )
            {
               ActionComboBox.SelectionChangeCommitted       += new EventHandler ( ActionComboBox_SelectionChangeCommitted ) ;
               MouseButtonsComboBox.SelectionChangeCommitted += new EventHandler ( MouseButtonsComboBox_SelectionChangeCommitted ) ;
               FeautreIdTextBox.Validating                   += new CancelEventHandler ( FeautreIdTextBox_Validating ) ;
               AddActionButton.Click                         += new EventHandler ( AddActionButton_Click ) ;
               RemoveActionButton.Click                      += new EventHandler ( RemoveActionButton_Click ) ;
               BrowseButton.Click                            += new EventHandler ( BrowseButton_Click ) ;
               AlternativeImageButton.Click                  += new EventHandler ( AlternativeImageButton_Click ) ;
            }

         #endregion
         
         #region Properties
            
         #endregion
         
         #region Events
         
            void ActionComboBox_SelectionChangeCommitted ( object sender, EventArgs e )
            {
               try
               {
                  if ( null != SelectedActionChanged ) 
                  {
                     SelectedActionChanged ( this, e ) ;
                  }
               }
               catch ( Exception exception ) 
               {
                  Messager.ShowError ( this, exception ) ;
               }
            }
            
            void MouseButtonsComboBox_SelectionChangeCommitted ( object sender, EventArgs e )
            {
               try
               {
                  if ( null != SelectedActionMouseButtonChanged ) 
                  {
                     SelectedActionMouseButtonChanged ( this, e ) ;
                  }
               }
               catch ( Exception exception ) 
               {
                  Messager.ShowError ( this, exception ) ;
               }
            }
            
            void FeautreIdTextBox_Validating(object sender, CancelEventArgs e)
            {
               try
               {
                  if ( null != ValidateFeatureId ) 
                  {
                     ValidateFeatureId ( this, e ) ;
                  }
               }
               catch ( Exception exception ) 
               {
                  e.Cancel = true ;
                  
                  Messager.ShowError ( this, exception ) ;
               }
            }
            
            void AddActionButton_Click(object sender, EventArgs e)
            {
               try
               {
                  if ( null != AddSelectedAction ) 
                  {
                     AddSelectedAction ( this, e ) ;
                  }
               }
               catch ( Exception exception ) 
               {
                  Messager.ShowError ( this, exception ) ;
               }
            }
            
            void RemoveActionButton_Click(object sender, EventArgs e)
            {
               try
               {
                  if ( null != RemoveSelectedAction ) 
                  {
                     RemoveSelectedAction ( this, e ) ;
                  }
               }
               catch ( Exception exception ) 
               {
                  Messager.ShowError ( this, exception ) ;
               }
            }
            
            void BrowseButton_Click ( object sender, EventArgs e )
            {
               try
               {
                  using ( OpenFileDialog openFileDlg = new OpenFileDialog ( ) )
                  {
                     openFileDlg.Filter = "Image Formats (*.jpg, *.bmp, *.ico)|*.jpg;*.bmp;*.ico" ;
                     
                     if ( openFileDlg.ShowDialog ( ) == DialogResult.OK ) 
                     {
                        ImagePictureBox.Image = Image.FromFile ( openFileDlg.FileName ) ;
                     }
                  }
               }
               catch ( Exception exception ) 
               {
                  Messager.ShowError ( this, exception ) ;
               }
            }
            
            void AlternativeImageButton_Click(object sender, EventArgs e)
            {
               try
               {
                  using ( OpenFileDialog openFileDlg = new OpenFileDialog ( ) )
                  {
                     openFileDlg.Filter = "Image Formats (*.jpg, *.bmp, *.ico)|*.jpg;*.bmp;*.ico" ;
                     
                     if ( openFileDlg.ShowDialog ( ) == DialogResult.OK ) 
                     {
                        AltImagePictureBox.Image = Image.FromFile ( openFileDlg.FileName ) ;
                     }
                  }
               }
               catch ( Exception exception ) 
               {
                  Messager.ShowError ( this, exception ) ;
               }
            }
            
         #endregion
         
         #region Data Members
            
         #endregion
         
         #region Data Types Definition
            
         #endregion
         
      #endregion
      
      #region internal
         
         #region Methods
            
         #endregion
         
         #region Properties
            
         #endregion
         
         #region Events
            
         #endregion
         
         #region Data Types Definition
            
         #endregion
         
         #region Callbacks
            
         #endregion
         
      #endregion
   }
}
